### PR TITLE
Airbyte v40

### LINKED ISSUE #197: Update canned connections to use Overwrite instead of Append sync mode
**Describe the bug**
The Faros Destination does not follow the sync mode spec (see https://github.com/faros-ai/airbyte-connectors/issues/575). Hence, all sync mode right now are either `Full Refresh | Append` or `Incremental | Append` which works as intended but not as an Airbyte user might expect. Once the Faros Destination is fixed, we will have to update the connections to use `Overwrite`.

### LINKED ISSUE #288: Cannot refresh schema on canned connections
**Describe the bug**
Airbyte fails to fetch an updated schema for canned connections. It works fine if one create a NEW source, link it to the EXISTING destination.

This is likely caused by one of 2 mismatches: our canned source def mentions [GitHub 0.3.8](https://github.com/faros-ai/faros-community-edition/blob/465b02a7e3c4c2a4b161d81964bdf37fdd76825c/init/resources/airbyte/workspace/airbyte_config/STANDARD_SOURCE_DEFINITION.yaml#L809) but:
* Airbyte changes it to 0.2.42, likely because of a version matching [thing](https://github.com/airbytehq/airbyte/blob/v0.39.37-alpha/airbyte-config/init/src/main/resources/seed/source_definitions.yaml#L306).
* It is possible that the source config [we pass](https://github.com/faros-ai/faros-community-edition/blob/465b02a7e3c4c2a4b161d81964bdf37fdd76825c/init/resources/airbyte/workspace/airbyte_config/SOURCE_CONNECTION.yaml#L2) is NOT 0.3.8 (and also NOT 0.2.42) - I can tell because the source config screen looks different between the canned one and when you create a new GitHub source

Suggested course of action: standardize on 0.2.42 for now, and redo the source config. 

Or, move to airbyte 40+ and redo the init.

**To Reproduce**
1. Start Faros from scratch
2. Put valid settings in the [EXISTING GitHub source](http://localhost:8000/workspaces/b740a9b4-048b-4500-b41b-038e14a3f2d5/source/5d9079ca-8173-406f-bfdb-41f19c62daff)
3. Go to its [connection](http://localhost:8000/workspaces/b740a9b4-048b-4500-b41b-038e14a3f2d5/connections/6421df4e-0c5a-4666-a530-9c01de683518/replication)
4. Click on refresh schema

Note that:
* step 2 is important, as Airbyte will refuse to update schema if the source has not been marked as "discoverable", which happens after a check if I am not mistaken
* this behavior happens on the Jira canned source as well; I suspect it happens on ALL canned sources  

**Expected behavior**
All streams are displayed.

**Screenshots and logs**
Worker is fine:
```
2023-03-03 17:10:30 2023-03-04 01:10:30 INFO i.a.w.t.TemporalAttemptExecution(get):108 - Docker volume job log path: /tmp/workspace/c74afff5-d342-47bc-b2e3-9884381a1ec3/0/logs.log
2023-03-03 17:10:30 2023-03-04 01:10:30 INFO i.a.w.t.TemporalAttemptExecution(get):113 - Executing worker wrapper. Airbyte version: 0.39.37-alpha
2023-03-03 17:10:30 2023-03-04 01:10:30 INFO i.a.c.i.LineGobbler(voidCall):82 - Checking if airbyte/source-github:0.2.42 exists...
2023-03-03 17:10:30 2023-03-04 01:10:30 INFO i.a.c.i.LineGobbler(voidCall):82 - airbyte/source-github:0.2.42 was found locally.
2023-03-03 17:10:30 2023-03-04 01:10:30 INFO i.a.w.p.DockerProcessFactory(create):108 - Creating docker job ID: c74afff5-d342-47bc-b2e3-9884381a1ec3
2023-03-03 17:10:30 2023-03-04 01:10:30 INFO i.a.w.p.DockerProcessFactory(create):163 - Preparing command: docker run --rm --init -i -w /data/c74afff5-d342-47bc-b2e3-9884381a1ec3/0 --log-driver none --name source-github-discover-c74afff5-d342-47bc-b2e3-9884381a1ec3-0-naayk --network host -v airbyte_workspace:/data -v /tmp/airbyte_local:/local -e DEPLOYMENT_MODE=OSS -e USE_STREAM_CAPABLE_STATE=false -e AIRBYTE_ROLE= -e WORKER_ENVIRONMENT=DOCKER -e WORKER_JOB_ATTEMPT=0 -e WORKER_CONNECTOR_IMAGE=airbyte/source-github:0.2.42 -e AIRBYTE_VERSION=0.39.37-alpha -e WORKER_JOB_ID=c74afff5-d342-47bc-b2e3-9884381a1ec3 airbyte/source-github:0.2.42 discover --config source_config.json
2023-03-03 17:10:32 2023-03-04 01:10:32 INFO i.a.w.t.TemporalAttemptExecution(get):134 - Stopping cancellation check scheduling...
```
but server complains:
```
2023-03-03 16:50:09 2023-03-04 00:50:09 ERROR i.a.s.e.UncaughtExceptionMapper(toResponse):22 - Uncaught exception
2023-03-03 16:50:09 java.lang.IllegalArgumentException: malformed JsonSchema object type, must have one of the following fields: properties, oneOf, allOf, anyOf in {}
2023-03-03 16:50:09     at io.airbyte.commons.json.JsonSchemas.traverseJsonSchemaInternal(JsonSchemas.java:213) ~[io.airbyte-airbyte-commons-0.39.37-alpha.jar:?]
2023-03-03 16:50:09     at io.airbyte.commons.json.JsonSchemas.traverseJsonSchemaInternal(JsonSchemas.java:194) ~[io.airbyte-airbyte-commons-0.39.37-alpha.jar:?]
2023-03-03 16:50:09     at io.airbyte.commons.json.JsonSchemas.traverseJsonSchemaInternal(JsonSchemas.java:206) ~[io.airbyte-airbyte-commons-0.39.37-alpha.jar:?]
2023-03-03 16:50:09     at io.airbyte.commons.json.JsonSchemas.traverseJsonSchema(JsonSchemas.java:107) ~[io.airbyte-airbyte-commons-0.39.37-alpha.jar:?]
2023-03-03 16:50:09     at io.airbyte.commons.json.JsonSchemas.traverseJsonSchemaWithFilteredCollector(JsonSchemas.java:142) ~[io.airbyte-airbyte-commons-0.39.37-alpha.jar:?]
2023-03-03 16:50:09     at io.airbyte.commons.json.JsonSchemas.traverseJsonSchemaWithCollector(JsonSchemas.java:125) ~[io.airbyte-airbyte-commons-0.39.37-alpha.jar:?]
2023-03-03 16:50:09     at io.airbyte.protocol.models.CatalogHelpers.getFullyQualifiedFieldNamesWithTypes(CatalogHelpers.java:224) ~[io.airbyte.airbyte-protocol-protocol-models-0.39.37-alpha.jar:?]
2023-03-03 16:50:09     at io.airbyte.protocol.models.CatalogHelpers.getStreamDiff(CatalogHelpers.java:320) ~[io.airbyte.airbyte-protocol-protocol-models-0.39.37-alpha.jar:?]
2023-03-03 16:50:09     at io.airbyte.protocol.models.CatalogHelpers.lambda$getCatalogDiff$7(CatalogHelpers.java:309) ~[io.airbyte.airbyte-protocol-protocol-models-0.39.37-alpha.jar:?]
2023-03-03 16:50:09     at java.lang.Iterable.forEach(Iterable.java:75) ~[?:?]
2023-03-03 16:50:09     at io.airbyte.protocol.models.CatalogHelpers.getCatalogDiff(CatalogHelpers.java:305) ~[io.airbyte.airbyte-protocol-protocol-models-0.39.37-alpha.jar:?]
2023-03-03 16:50:09     at io.airbyte.server.handlers.ConnectionsHandler.getDiff(ConnectionsHandler.java:263) ~[io.airbyte-airbyte-server-0.39.37-alpha.jar:?]
2023-03-03 16:50:09     at io.airbyte.server.handlers.WebBackendConnectionsHandler.webBackendGetConnection(WebBackendConnectionsHandler.java:261) ~[io.airbyte-airbyte-server-0.39.37-alpha.jar:?]
2023-03-03 16:50:09     at io.airbyte.server.apis.ConfigurationApi.lambda$webBackendGetConnection$88(ConfigurationApi.java:812) ~[io.airbyte-airbyte-server-0.39.37-alpha.jar:?]
2023-03-03 16:50:09     at io.airbyte.server.apis.ConfigurationApi.execute(ConfigurationApi.java:873) ~[io.airbyte-airbyte-server-0.39.37-alpha.jar:?]
2023-03-03 16:50:09     at io.airbyte.server.apis.ConfigurationApi.webBackendGetConnection(ConfigurationApi.java:812) ~[io.airbyte-airbyte-server-0.39.37-alpha.jar:?]
```

### PR DESCRIPTION
**Pull request by thomas-gerber**
# Description
Moving to Airbyte 0.40.9
Also:
Fixes #197 
Fixes #288  

TODO:
- [x] Add the other sources besides the main 4
- [x] documentation links need to be changed as the IDs are not fixed anymore
- [x] migration strategy
- docker extension upgrade (separate PR, once the new init is released

## Type of change
- Bug fix (non-breaking change which fixes an issue)
- New feature (non-breaking change which adds functionality)
- Breaking change (fix or feature that would cause existing functionality to not work as expected)
- This change requires a documentation update

# Migration
A simple restart will upgrade users to 0.40.9. Note that those migrated users will remain on their existing sources and Faros destination versions (until we re-enable auto-update).
This was tested the following way:
1. start an older version of CE (Airbyte v0.39.37-alpha)
2. sync Jira and GitHub data through the CLI
3. stop that older Faros, but do not delete the volumes
4. start a newer version of CE (this branch, on Airbyte v0.40.9)
5. do an incremental sync on those 2 sources

# Ability to update source schema for canned sources
It will work *if the starting version of CE is after this merge*. Older users of Faros still have to recreate the sources and connections.
This was tested the following way:
1. start a newer version of CE
2. configure the GitHub source and sync it
3. in the connection replication settings, refresh source schema
4. one sees all the streams we haven't configured as new, and all existing streams as having been modified
5. enable pull_request_stats and save
6. sync again, and that new stream catches up

# A note on sync mode
We either use Incremental with append_dedup if incremental is available source-side, full_refresh with overwrite otherwise. Note that we don't fully abide by the spec, as we do NOT delete the data for that stream in the overwrite mode. Unfortunately, there is no sync mode that describes that behavior ATM.

Note that all existing streams are marked as modified only because we didn't pass along the json schema during canned setup, which is now captured for the first time

# Checklist
- [x] Have you checked to there aren't other open Pull Requests for the same update/change?
- [x] Have you lint your code locally before submission?
- [x] Have you added an explanation of what your changes do and why you'd like us to include them?
- [x] Have you successfully run tests with your changes locally?

### CODE REVIEW

File: .env [hunk 1]
```diff
@@ -118,13 +118,17 @@ API_URL=/api/v1/
 
 ### JOBS ###
 # Relevant to scaling.
-SYNC_JOB_MAX_ATTEMPTS=1
-SYNC_JOB_MAX_TIMEOUT_DAYS=1
+SYNC_JOB_MAX_ATTEMPTS=3
+SYNC_JOB_MAX_TIMEOUT_DAYS=3
```
Comment (ogusak) [inline .env:122]: This would allow running job for 3 days - likely too long for real usecases

File: docker-extension/.env [hunk 2]
```diff
@@ -118,13 +118,17 @@ API_URL=/api/v1/
 
 ### JOBS ###
 # Relevant to scaling.
-SYNC_JOB_MAX_ATTEMPTS=1
-SYNC_JOB_MAX_TIMEOUT_DAYS=1
+SYNC_JOB_MAX_ATTEMPTS=3
+SYNC_JOB_MAX_TIMEOUT_DAYS=3
```
Comment (ogusak) [inline docker-extension/.env:122]: similar to earlier comment - 3 days run may be too long

**Comment by ogusak** [comment]
If we plan on upgrading k8s deploy - [this](https://github.com/faros-ai/faros-community-edition/blob/main/kube/base/kustomization.yaml#L5) needs to be changed as well

**Comment by thomas-gerber** [comment]
> If we plan on upgrading k8s deploy - [this](https://github.com/faros-ai/faros-community-edition/blob/main/kube/base/kustomization.yaml#L5) needs to be changed as well

Fixed

**Review by ogusak — APPROVED** [review]

File: init/package.json [hunk 3]
```diff
@@ -23,13 +23,16 @@
     "resources/"
   ],
   "dependencies": {
+    "@types/js-yaml": "^4.0.5",
```
Comment (cjwooo) [inline init/package.json:26]: `@types/*` libraries should be devDependencies only.

File: .github/workflows/pr.yml [hunk 4]
```diff
@@ -53,9 +53,34 @@ jobs:
         run: npm run test:unit -- --coverage
         working-directory: init
 
-      # In order to optimize for speed, we use the latest init image.
+      - name: Docker meta
```
Comment (ypc-faros) [inline .github/workflows/pr.yml:56]: Back in https://github.com/faros-ai/faros-community-edition/pull/139 , we agreed to only build from source in the release workflow. Is there a specific reason we're going back here?

Comment (ypc-faros) [reply .github/workflows/pr.yml:56]: Sounds good, thanks!

File: init/src/airbyte/initv40.ts [hunk 5]
```diff
@@ -0,0 +1,520 @@
...
+    );
+    await this.createAndConnectSource(
+      'Jira',
+      workspaceId,
+      farosDestinationId,
+      yamlSourceData,
+      yamlCatalogData,
+      jiraSourceDefinitionId
+    );
+    await this.handleFarosSource(
```
Comment (cjwooo) [inline init/src/airbyte/initv40.ts:423]: These `handleFarosSource` calls can be consolidated:

```
const sources = ['Bitbucket', 'Phabricator', ...]
for (const source of sources) {
  await this.handleFarosSource(source, workspaceId, ...)
}
```

Comment (thomas-gerber) [reply init/src/airbyte/initv40.ts]: duh; thanks, done!

**Review by cjwooo — APPROVED** [review]

**Pull request merged**